Bugfix/update

JMilamber opened the pull request:
1.5.11 - 
- Add Golem Hp to Config
- Re-Sort golem progression, with core and nether towers switching places.
    Old: 
    Land->Ocean->Nether->Core->End->Sky

    New:
    Land->Ocean>Core->Nether->End->Sky

1.5.12 -
- Added updated monolith textures. 
- Removed Totem Block

1.5.13 -
- Give chests Explosion Resistance

1.5.14 -
- Fixed bug where Tower would start crumbling when golem was pushed/moved to just outside of the tower (before it would teleport back) 
- Remove Sky-Tower from structure registry completely

1.5.15 -
- Fixed sounds still playing for player when dying and respawning far away
- Fixed title commands not working in single-player without cheats sometimes. 
    Title commands are now sent as if by the server and are more stable.

JMilamber committed:
Added config option for Golem HP, re-ordered golems so that it goes core->Nether->End

JMilamber committed:
Added fixed textures for Core, Nether, and End monoloiths

JMilamber committed:
remove TotemBlock, add Explosion resistance to chests

JMilamber committed:
Give land tower chest half the explosion resistance, so that there is a reason to collect the chest.

JMilamber committed:
Change Tower Chest explosion resistance to that of blackstone (6.0) so that it is possible to destroy nt much harder to do so.

Xratedjunior commented on src/main/java/com/BrassAmber/ba_bt/entity/hostile/golem/BTGolemEntity.java:
```diff
@@ -18,4 +22,10 @@ protected void registerGoals() {
 	}
+
+	/*********************************************************** Properties ********************************************************/
+
```
Feels like we need an 'Override' here to make it clear?

JMilamber replied:
Done :)

JMilamber committed:
Fixed bug where Tower would start to crumble when golem was outside of tower but not outside of reset range.

Hopefully fixed a sound bug where sounds would continue to play at players location after dying and moving hundreds of blocks away

set Golem base health to 250 and minimum to 200, changed golem damage from 8.5/15/22.5 to 6/10/15

Removed Totem block entirely

JMilamber committed:
Actually finally remove SkyTower from in-game locate registry

JMilamber commented:
@Xratedjunior  
If you could test the sound bug mentioned for if this fixes them that would be awesome, a little short on time atm doing the 1.18.1 update. 
I only tested for the golem/destroy tower one mentioned.

Xratedjunior commented:
You mean that crash?

JMilamber commented:
No, that one hasn't given an actual data on when/why/what happened. 

On the Curseforge someone commented and notified me that when the golem left the radius of the tower (but not far enough to be teleported back) that the tower crumble would start. 
This was because of how I was checking to see if the golem was still alive. Ai was using getNearestEntity which works fine until I remembered all of the mobs that spawn just below the golem floor. So I changed how it checks for the golem and it works correctly now. 

The user also mentioned that if you kill the golem, but then die and respawn far away you continue to hear the sound effects for a bit. 

I changed the function calling the sound event to level.playLocalSound but didn't get to test if the change fixes the problem.

JMilamber committed:
Figured out new better way to send commands that will work whether the player is opped or not, started reworiking to use it, not finished yet and still needs testing afterwards.

JMilamber committed:
Updated all title commands in DestroyTowerEntity to be sent by the server rather than each player individually. No longer requires op powers to see titles.

JMilamber committed:
Remove last totem asset and update vserion number

JMilamber committed:
Revised txt files in base folder- specifically Inspiration file. Also forgot to removed distance modifier for some title in last commit, fixed now

JMilamber committed:
Upped distance from tower that p[layer can be while tower is crumbling without it stopping the destruction process from 5 to 125.
